from_dict/_from_dict.py

```diff
@@ -88,8 +88,19 @@ def get_constructor_type_hints(
     if cls is None:
         return None
 
-    return typing.get_type_hints(cls.__init__, global_ns, local_ns) or typing.get_type_hints(cls, global_ns, local_ns)
+    # Can't use caching if a namespace dictionary is passed in.
```

Comment:
Why can't we? Is it because we're passing in `dict` and they cannot be hashed? Have you considered using `FrozenDict` from [frozendict](https://pypi.org/project/frozendict/) instead? They are hashable. So we could do something like:

```
@functools.lru_cache(max_size=1000)
def _get_constructor_type_hints(cls: Type, global_ns: Optional[FrozenDict], local_ns: Optional[FrozenDict]) -> Optional[Mapping[str, Type]]:
```

And then convert the incoming namespaces into frozendicts when we receive them.

Reply:
I was think passing in global_ns and local_ns was such a small use case that it was not optimizing.

Whenever I passed in a value for global_ns or local_ns, I just called ```globals()``` or ```locals()```. Converting them to ```FrozenDict``` might undo an performance improvement that caching gives us and would require the client to have another dependency they don't want. I will investigate the performance impact. 

I will look into ways to make caching work with the namespace parameters. Feel free to over any other suggestions.